### PR TITLE
Add ability to specify a CA certificate

### PR DESCRIPTION
**Pull request by Infra-Red**
Allow the resource to accept additional trusted certificates by appending our self-signed certs to a copy of the host trusted CA certs.

Resolves https://github.com/concourse/registry-image-resource/issues/240.

### CODE REVIEW

**Comment by vito** [comment]
Woops, sorry we didn't see this! For some reason our PR overview tool didn't pick it up. 🤔 Leaving this comment to kick it again - we'll assign it to someone on Monday if no one reviews it before then! Thanks for your work, this is a big missing piece.

**Comment by vito** [comment]
Ah - `PayloadTooLargeError: request entity too large` handling the webhook, probably because of that giant code block. That's a fun edge case. I removed it from the description since we should be able to see the same output in the PR builds anyhow. Hope you don't mind. 👍

File: commands/out.go [hunk 1]
```diff
@@ -245,11 +256,47 @@ func createAuth(req resource.OutRequest) *authn.Basic {
 	}
 }
 
+func createTransport(req resource.OutRequest) (http.RoundTripper, error) {
```
Comment (vito) [inline commands/out.go:259]: Maybe we can extract this into a method on `Source`? It looks to be duplicated at the moment with `types.go`.

Actually not sure why the code doesn't just use the `(Source).AuthOptions` method here too, but I haven't looked too closely. 🤔

**Review by vito — CHANGES_REQUESTED** [review]
This is a good start - thanks for also updating the README! It'll need some tests before we can call it done though.

I think we could run a fake registry over HTTPS and show that basic resource `in`, `out`, and `check` calls fail without the cert and succeed with the cert.

We already run a fake registry for the `out` tests so maybe you could look to that for an example. For running a TLS server, you can use [`gomega.NewTLSServer`](https://pkg.go.dev/github.com/onsi/gomega@v1.10.5/ghttp#NewTLSServer) and then access the certificate using [(*httptest.Server).Certificate](https://golang.org/pkg/net/http/httptest/#Server.Certificate) to pass it along to the resource. It's self-signed, so you should be able to use it as a CA.

File: commands/out.go (hunk 1 above)
Comment (Infra-Red) [reply commands/out.go]: I updated the `out` command to use a method from `Source`.

File: commands/out.go [hunk 2]
```diff
@@ -248,8 +258,15 @@ func createAuth(req resource.OutRequest) *authn.Basic {
 func aliasesToBump(req resource.OutRequest, repo name.Repository, ver *semver.Version) ([]name.Tag, error) {
 	variant := req.Source.Variant
 
-	opts := []remote.Option{}
-	opts = append(opts, remote.WithAuth(createAuth(req)))
```
Comment (vito) [inline commands/out.go:252]: Can this `createAuth` function be removed now?

**Review by vito — CHANGES_REQUESTED** [review]
Thanks, this looks great! Just nits to pick from here. I do think those new `It`s are a bit too noisy, but thanks for the due diligence. 🙂

File: commands/out.go (hunk 2 above)
Comment (Infra-Red) [reply commands/out.go:252]: `createAuth ` is still used in the `signImages` function. The `gcr.NewTrustedGcrRepository()` function requires authorization to be passed.

File: check_test.go [hunk 3]
```diff
@@ -67,6 +70,10 @@ var _ = Describe("Check", func() {
 				req.Version = nil
 			})
 
+			It("should not error", func() {
```
[the remark this answers is not in the document]
Comment (Infra-Red) [reply check_test.go]: I added this assertion to existing `It` blocks, I think this is the most common pattern. Fixed in https://github.com/concourse/registry-image-resource/pull/264/commits/40734d20866202465015bbdc37c337542cc5ae2c.

File: in_test.go [hunk 4]
```diff
@@ -86,6 +90,10 @@ var _ = Describe("In", func() {
 			}
 		})
 
+		It("should not error", func() {
```
[the remark this answers is not in the document]
Comment (Infra-Red) [reply in_test.go]: Fixed in https://github.com/concourse/registry-image-resource/pull/264/commits/40734d20866202465015bbdc37c337542cc5ae2c.

**Review by vito — APPROVED** [review]
LGTM - thanks again!

**Pull request merged**